cider/function/FunctionLookup.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_FUNCTION_FUNCTIONLOOKUP_H
+#define CIDER_FUNCTION_FUNCTIONLOOKUP_H
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "function/SubstraitFunctionCiderMappings.h"
+#include "function/substrait/SubstraitFunctionLookup.h"
+#include "function/substrait/SubstraitType.h"
+#include "function/substrait/VeloxToSubstraitMappings.h"
+
+enum PlatformType { SubstraitPlatform, PrestoPlatform, SparkPlatform };
+
+struct FunctionSignature {
+  std::string func_name;
+  std::vector<cider::function::substrait::SubstraitTypePtr> arguments;
+  cider::function::substrait::SubstraitTypePtr return_type;
+  PlatformType from_platform;
+};
+
+struct FunctionDescriptor {
+  FunctionSignature func_sig;
+  SQLOps scalar_op_type = SQLOps::kUNDEFINED_OP;
+  SQLAgg agg_op_type = SQLAgg::kUNDEFINED_AGG;
+  OpSupportExprType op_support_expr_type = OpSupportExprType::kUNDEFINED_EXPR;
+};
+
+using FunctionDescriptorPtr = std::shared_ptr<FunctionDescriptor>;
+using SubstraitFunctionLookupPtr =
+    std::shared_ptr<const cider::function::substrait::SubstraitFunctionLookup>;
+
+class FunctionLookup {
```

Comment:
rename this class? `FunctionLookupEngine` or `FunctionStore` or something else?

Reply:
Done. Use FunctionLookupEngine.